src/snowcli/snow_connector.py

```diff
@@ -56,3 +86,46 @@ def _find_command_path():
         # Example: SNOWCLI.WAREHOUSE.STATUS
         return ".".join(["SNOWCLI", *ctx.command_path.split(" ")[1:]]).upper()
     return "SNOWCLI"
+
+
+def load_pem_to_der(private_key_path: str) -> bytes:
+    """
+    Given a private key file path (in PEM format), decode key data into DER
+    format
+    """
+    with open(private_key_path, "rb") as f:
+        private_key_pem = f.read()
+
+    private_key_passphrase = os.getenv("PRIVATE_KEY_PASSPHRASE", None)
+    if (
+        private_key_pem.startswith(ENCRYPTED_PKCS8_PK_HEADER)
+        and private_key_passphrase is None
+    ):
+        raise Exception(
+            "Encrypted private key, you must provide the"
+            "passphrase in the environment variable PRIVATE_KEY_PASSPHRASE"
+        )
```

Comment:
Correct me if I'm wrong, but we can throw this exception if PRIVATE_KEY_PASSPHRASE variable is not set, so we can do this validation before reading the file to potentially fail faster

Reply:
Users can use an unencrypted key, so we have to read the file to know if it requires the passphrase.